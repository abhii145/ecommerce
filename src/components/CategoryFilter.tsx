interface Category {
  slug: string
  name: string
}

interface CategoryFilterProps {
  categories: Category[]
  selectedCategory: string
  onCategoryChange: (category: string) => void
}

const CategoryFilter: React.FC<CategoryFilterProps> = ({
  categories,
  selectedCategory,
  onCategoryChange,
}) => {
  return (
    <div className="flex items-center space-x-2">
      <label htmlFor="category" className="font-semibold text-gray-700">
        Category:
      </label>
      <select
        id="category"
        value={selectedCategory}
        onChange={(e) => onCategoryChange(e.target.value)}
        className="border border-gray-300 p-2 rounded-md"
        aria-label="Select category"
      >
        <option value="">All Categories</option>
        {categories.map((category) => (
          <option key={category.slug} value={category.slug}>
            {category.name}
          </option>
        ))}
      </select>
    </div>
  )
}

export default CategoryFilter
