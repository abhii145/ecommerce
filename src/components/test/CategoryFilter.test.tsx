import { render, screen, fireEvent } from "@testing-library/react"
import CategoryFilter from "../CategoryFilter"

const categories = [
  { slug: "electronics", name: "Electronics" },
  { slug: "fashion", name: "Fashion" },
]

describe("CategoryFilter Component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <CategoryFilter
        categories={categories}
        selectedCategory=""
        onCategoryChange={() => {}}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders CategoryFilter component", () => {
    render(
      <CategoryFilter
        categories={categories}
        selectedCategory=""
        onCategoryChange={() => {}}
      />
    )

    expect(screen.getByLabelText(/Category/i)).toBeInTheDocument()
    expect(screen.getByText(/All Categories/i)).toBeInTheDocument()
    expect(screen.getByText(/Electronics/i)).toBeInTheDocument()
    expect(screen.getByText(/Fashion/i)).toBeInTheDocument()
  })

  it("calls onCategoryChange when a category is selected", () => {
    const onCategoryChange = jest.fn()
    render(
      <CategoryFilter
        categories={categories}
        selectedCategory=""
        onCategoryChange={onCategoryChange}
      />
    )

    fireEvent.change(screen.getByLabelText(/Category/i), {
      target: { value: "electronics" },
    })
    expect(onCategoryChange).toHaveBeenCalledWith("electronics")
  })
})
