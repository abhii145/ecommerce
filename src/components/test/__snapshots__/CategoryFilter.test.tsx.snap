// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFilter Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center space-x-2"
  >
    <label
      class="font-semibold text-gray-700"
      for="category"
    >
      Category:
    </label>
    <select
      aria-label="Select category"
      class="border border-gray-300 p-2 rounded-md"
      id="category"
    >
      <option
        value=""
      >
        All Categories
      </option>
      <option
        value="electronics"
      >
        Electronics
      </option>
      <option
        value="fashion"
      >
        Fashion
      </option>
    </select>
  </div>
</DocumentFragment>
`;
